### ISSUE TITLE
Copying images shared from a phone crashes Gnome Shell

### ISSUE THREAD

**Issue by danielbutler65** [oae/gnome-shell-pano#322]
## Description

When copying phone sourced images, pano is crashing the Gnome Shell. 

### Problem Explanation

Write a clear and concise description of what the bug is:

The bug happens when copying images taken from a phone shared with teams and Signal desktop at least, probably more. The PNG created in the pano database location before the crash is significantly larger than the source image

### Expected Behavior

- Pano shouldn't crash the Gnome Shell when copying images

## Reproduction

Copy a phone image to the clipboard

### Steps To Reproduce

Steps to reproduce, if applicable:

- I have seen this when copying a large image from Microsoft Teams or Signal Desktop, there is a good chance it doesn't matter what the source is. 

### Details

Mark with [ ] all that applies:

#### It happens with any application?

- [x] Yes, it applies to any application.
- [x] No. Only with the following applications:
  - Signal desktop
  - Microsoft Teams
- [ ] It works with the following applications that I have tried:

#### It happens only on one computer?

- [x] I don't know.
- [ ] No. I have tried it on more than one computer.

#### It happens only with some specific gnome configuration?

- [x] I don't think that the configuration matters.
- [ ] Yes. Only if the following config is set up:
  - Fill in a list with any configuration tha applies.

#### It happens only with some specific extension installed?

- [x] I don't think that the installed extensions affect the bug/behavior.
- [ ] Yes. Only if the following gnome extension is installed:

### Environment

- GNOME Shell version: 42.9
- Distro: Ubuntu
- Distro version: 22.04.5

#### Screenshot 1 - Image being copied out of teams
![image](https://github.com/user-attachments/assets/08fc7665-97b3-4d22-a12d-6aaeb857b2b7)

##### The source image being copied
![OffendingImageFromTeams](https://github.com/user-attachments/assets/4a4184d5-4098-4bd6-9ea7-fbc283e84ae5)

#### Pano Application Version
19

#### Gnome

``` 
Oct 11 11:17:25 L106153 gnome-shell[12604]: GNOME Shell crashed with signal 11
Oct 11 11:17:25 L106153 gnome-shell[12604]: == Stack trace for context 0x5a2a774254a0 ==
Oct 11 11:17:25 L106153 audit[12604]: ANOM_ABEND auid=1000 uid=1000 gid=1000 ses=7 subj=unconfined pid=12604 comm="gnome-shell" exe="/usr/bin/gnome-shell" sig=11 res=1
Oct 11 11:17:25 L106153 gnome-shell[12604]: #0   7ffea5cf79f0 b   /home/user/.local/share/gnome-shell/extensions/pano@elhan.io/extension.js:330 (8ce7d065dd0 @ 274)
Oct 11 11:17:25 L106153 gnome-shell[12604]: #1   7ffea5cf81b0 b   /home/user/.local/share/gnome-shell/extensions/pano@elhan.io/extension.js:2236 (8ce7d071a10 @ 261)
Oct 11 11:17:25 L106153 gnome-shell[12604]: #2   7ffea5cf8c00 b   /home/user/.local/share/gnome-shell/extensions/pano@elhan.io/extension.js:2233 (8ce7d0719c0 @ 24)
Oct 11 11:17:25 L106153 gnome-shell[12604]: #3   5a2a7d91fac0 i   /home/user/.local/share/gnome-shell/extensions/pano@elhan.io/extension.js:2167 (8ce7d071830 @ 244)
Oct 11 11:17:25 L106153 gnome-shell[12604]: #4   5a2a7d91fa18 i   self-hosted:689 (22d38ff07510 @ 15)
```

#### Pano Configuration

**Command:** `dconf dump /org/gnome/shell/extensions/pano/`

``` 
command returns nothing, no settings have been changed from the default values
```

#### Enabled Extensions

``` 
['apps-menu@gnome-shell-extensions.gcampax.github.com',
'drive-menu@gnome-shell-extensions.gcampax.github.com',
'auto-move-windows@gnome-shell-extensions.gcampax.github.com',
'pano@elhan.io',
'places-menu@gnome-shell-extensions.gcampax.github.com',
'KeepAwake@jepfa.de',
'autoselectheadset@josephlbarnett.github.com']
```

**Comment by danielbutler65** [comment]
This was a pretty much 100% occurrence rate for a few weeks but it was a little less immediately after posting this. I ended up clearing my cache and history and have not reproduced it since. I will keep trying but this can probably be closed out if I can't reproduce it again.

**Comment by danielbutler65** [comment]
This is still occurring

```Oct 15 10:53:20 L106153 gnome-shell[5260]: GNOME Shell crashed with signal 11
Oct 15 10:53:20 L106153 gnome-shell[5260]: == Stack trace for context 0x614c8823d490 ==
Oct 15 10:53:20 L106153 gnome-shell[5260]: #0   614c8c3a0c48 i   /home/user/.local/share/gnome-shell/extensions/pano@elhan.io/extension.js:330 (203ee0065dd0 @ 274)
Oct 15 10:53:20 L106153 gnome-shell[5260]: #1   614c8c3a0b38 i   /home/user/.local/share/gnome-shell/extensions/pano@elhan.io/extension.js:2236 (203ee0071a10 @ 261)
Oct 15 10:53:20 L106153 gnome-shell[5260]: #2   614c8c3a0aa8 i   /home/user/.local/share/gnome-shell/extensions/pano@elhan.io/extension.js:2233 (203ee00719c0 @ 24)
Oct 15 10:53:20 L106153 gnome-shell[5260]: #3   614c8c3a0a08 i   /home/user/.local/share/gnome-shell/extensions/pano@elhan.io/extension.js:2167 (203ee0071830 @ 244)
Oct 15 10:53:20 L106153 gnome-shell[5260]: #4   614c8c3a0968 i   self-hosted:1486 (20ff50d68100 @ 30)
Oct 15 10:53:20 L106153 gnome-shell[5260]: #5   7fff15b2e590 b   self-hosted:689 (1238ab604510 @ 15)
Oct 15 10:53:20 L106153 audit[5260]: ANOM_ABEND auid=1000 uid=1000 gid=1000 ses=3 subj=unconfined pid=5260 comm="gnome-shell" exe="/usr/bin/gnome-shell" sig=11 res=1
```

**Comment by Totto16** [comment]
Interesting, still with this image from above?

**Comment by danielbutler65** [comment]
> Interesting, still with this image from above?

Yep, I was able to crash it again copying that image out of the defect using Edge. 

One potentially interesting variable that may be affecting this is the presence of Microsoft Defender Endpoint on my machine, perhaps that is stalling something long enough during a large copy to cause issues...

https://learn.microsoft.com/en-us/defender-endpoint/linux-install-manually

**Comment by Totto16** [comment]
I'll test it, it's strange. It's going to be hard to reproduce it, since on my old really slow laptop, I also copied some large images and it never crashed, it took really long, but that was it 🤷🏼‍♂️
